_posts/2020-05-07-tuning-a-complex-graph/tuning-a-complex-graph.Rmd

```diff
@@ -0,0 +1,377 @@
+---
+title: "Tuning a complex graph"
+categories:
+  - tuning
+  - mlr3pipelines
+  - classification
+author:
+  - name: Lennart Schneider
+date: 05-07-2020
+description: |
+  We show how to tune a complex graph for a single task.
+output:
+  distill::distill_article:
+    self_contained: false
+    css: ../../custom.css
+bibliography: biblio.bib
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(
+  echo = TRUE,
+  R.options = list(width = 80)
+)
+library(mlr3book)
+```
+
+In this use case we show how to tune a rather complex graph consisting of different preprocessing steps and different learners where each preprocessing step and learner itself has parameters that can be tuned.
+You will learn the following:
+
+* Build a `r ref("Graph", "Graph")` that consists of two common preprocessing steps, then switches between two dimensionality reduction techniques followed by a `r ref("Learner", "Learner")` vs. no dimensionality reduction followed by another `r ref("Learner", "Learner")`
+* Define the search space
```

Comment:
* Define the search space for tuning that handles inter-dependencies between pipeline steps and hyperparameters